### PR TITLE
The helper functions that will be used for authentication

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
These helper functions will be used for authentication

Auth Middleware finalised and firebase token checked with the client side
User duplicacy issue fix implemented in user service and email removed from update user DTO

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

Terminal commands to test code:-
1). npm run lint
2). npm run lint:fix
3). nest start
4). npm run test

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09  These functions will be used for the authentication
Also when a new user will be created , this will be used to check whether the email already exists or not to avoid duplicacy of user creation.

File: src/middleware/tokenCheck.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      const { authToken } = req;
+      if (!authToken) {
+        const err = new ForbiddenException(
+          'Unauthorized - missing token in tokencheck',
+        );
+        next(err);
+        return;
+      }
+      const userInfo = await admin.auth().verifyIdToken(authToken);
+      req.body.owner = userInfo.uid;
```
Comment (Abhishek-kumar09) [inline src/middleware/tokenCheck.ts:27]: Owner is not the property here. 
It should be user

Comment (Devesh21700Kumar) [reply src/middleware/tokenCheck.ts]: oh, yeah 
changed that now

**Review by Devesh21700Kumar — COMMENTED** [review]
Auth Middleware finalised and firebase token checked with the client side
User duplicacy issue fix implemented in user service and email removed from update user DTO

File: src/main-firebase.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import * as express from 'express';
 import * as functions from 'firebase-functions';
 import * as admin from 'firebase-admin';
 
-admin.initializeApp();
+// admin.initializeApp();
```
Comment (Abhishek-kumar09) [inline src/main-firebase.ts:8]: Why this?

Comment (Devesh21700Kumar) [reply src/main-firebase.ts:8]: app was already initialised so did not need these
will remove it straight away

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**